contribs/gnodev/pkg/dev/node.go

```diff
@@ -173,26 +175,53 @@
 	int64BlockNum := int64(blockNum)
 	b, err := n.client.Block(&int64BlockNum)
 	if err != nil {
-		return []gnoland.TxWithMetadata{}, fmt.Errorf("unable to load block at height %d: %w", blockNum, err) // nothing to see here
+		return nil, fmt.Errorf("unable to load block at height %d: %w", blockNum, err)
 	}
+	txs := b.Block.Data.Txs
+
+	bres, err := n.client.BlockResults(&int64BlockNum)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load block at height %d: %w", blockNum, err)
+	}
+	deliverTxs := bres.Results.DeliverTxs
+
+	// Sanity check
+	if len(txs) != len(deliverTxs) {
+		panic(fmt.Errorf("invalid block txs len (%d) vs block result txs len (%d)",
+			len(txs), len(deliverTxs),
+		))
+	}
+
+	txResults := make([]*abci.ResponseDeliverTx, len(deliverTxs))
+	for i, tx := range deliverTxs {
+		txResults[i] = &tx
+	}
+
+	// XXX: Consider replacing a failed transaction with an empty transaction
```

Comment:
If you're counting the number of txs somewhere, you can easily swap this counter for the number of successful transactions, not total txs (which is true, because you're stepping through good ones only)